### PR TITLE
feature: cbi-726 test transfer all

### PR DESCRIPTION
**Pull request by huythanhcere**
Ticket ([url](https://cerenetwork.atlassian.net/browse/CBI-726))
Fix tests to confirm above.
Update add input fee for transfer all method.

### CODE REVIEW

**Review by evgeny-s — APPROVED** [review]
LGTM

File: cere02/lib.rs [hunk 1]
```diff
@@ -149,18 +149,18 @@ mod ddc {
         /// destination account can be the same as the contract owner
         /// return OK or an error
         #[ink(message)]
-        pub fn transfer_all_balance(&mut self, destination: AccountId) -> Result<()> {
+        pub fn transfer_all_balance(&mut self, destination: AccountId, transaction_fee: Balance) -> Result<()> {
```
Comment (naure) [inline cere02/lib.rs:152]: Where does this fee come from? Any documentation about that?

File: cere02/lib.rs [hunk 2]
```diff
@@ -149,18 +149,18 @@ mod ddc {
...
+
             let contract_bal = self.env().balance();
-            // ink! transfer emit a panic!, the function below doesn't work, at least with this nightly build
-            // self.env().transfer(destination, contract_bal).expect("pay out failure");
 
-            let _result = match self.env().transfer(destination, contract_bal) {
-                Err(_e) => Err(Error::TransferFailed),
-                Ok(_v) => Ok(()),
-            };
+            let _result = self.env().transfer(destination, contract_bal - transaction_fee);
```
Comment (naure) [inline cere02/lib.rs:163]: This does not check the result. There must be a `?` or an `unwrap` or `return Err`.

**Review by naure — CHANGES_REQUESTED** [review]
Please add a test where transfer_all_balance is called from another account which is not owner. See how other tests use `set_caller` to do that.

File: cere02/lib.rs (hunk 1 above)
Comment (huythanhcere) [reply cere02/lib.rs:152]: This comment out of scope of the ticket. It is right. I will fix in next commit then tag you.

**Comment by huythanhcere** [comment]
> Please add a test where transfer_all_balance is called from another account which is not owner. See how other tests use `set_caller` to do that.

This comment out of scope of the ticket. It is right. I will fix in next commit then tag you.

File: cere02/lib.rs (hunk 2 above)
Comment (huythanhcere) [reply cere02/lib.rs:163]: Result is unnecessary to check here. If error, token is not transferred.

Comment (huythanhcere) [reply cere02/lib.rs:163]: Anw, i will fix this.

File: cere02/lib.rs (hunk 1 above)
Comment (naure) [reply cere02/lib.rs:152]: Ok I understand now. This is not a transaction fee, but the deposit for the contract storage that must stay on the contract account.

**Review by naure — APPROVED** [review]

**Pull request merged**